### ISSUE TITLE
Kinesis Enhanced fanout consumer generates broken template

### ISSUE THREAD

**Issue by suparngp** [serverless/serverless#8404]
Kinesis Enhanced Fan out consumer setting `true` introduced in #7320 is broken on serverless version 2.7.0. As compared to v2.4.0, the generated policy document for a `consumer: true` setting is incorrect.
I am upgrading to the latest since another issue I had reported before #8328 is now resolved.

On 2.4.0, the output is 

```
                {
                  "Effect": "Allow",
                  "Action": [
                    "kinesis:SubscribeToShard"
                  ],
                  "Resource": [
                    {
                      "Ref": "....ArnConsumerStreamConsumer"
                    }
                  ]
                }
```
where as the same template produces 
```
               {
                  "Effect": "Allow",
                  "Action": [
                    "kinesis:SubscribeToShard"
                  ],
                  "Resource": [
                    "true"
                  ]
                }
```

The Resource "true" causes deployment to fail.

<details>
<summary><code>serverless.yml</code></summary>

```yaml
plugins:
  - serverless-webpack

custom:
  webpack:
    packager: pnpm
service: myservice
package:
  individually: true
provider:
  name: aws
  stage: api
  stackName: MyServiceStackLambda
  runtime: nodejs12.x
  region: us-east-1
  iamRoleStatements:
    - Effect: Allow
      Action:
        - "kinesis:Get*"
        - "kinesis:DescribeStream"
        - "kinesis:PutRecord"
        - "kinesis:PutRecords"
      Resource:
        - "Fn::ImportValue": MyServiceStack-Kinesis-events-Arn
functions:
  - myfunction:
      handler: src/handlers/index.handler
      events:
        - stream:
            type: kinesis
            arn:
              "Fn::ImportValue": MyServiceStack-Kinesis-events-Arn
            consumer: true
```

</details>

<details>
<summary><b><code>sls deploy</code> output</b></summary>

```
  An error occurred: IamRoleLambdaExecution - Resource true must be in ARN format or "*". (Service: AmazonIdentityManagement; Status Code: 400; Error Code: MalformedPolicyDocument; Request ID: e8086fcf-abba-4aed-9276-8d28af5edc99; Proxy: null).
```

</details>

<b>Installed version</b>

```
Framework Core: 2.7.0 (local)
Plugin: 4.1.1
SDK: 2.3.2
Components: 3.2.2
```

**Comment by suparngp** [comment]
Also, I tested 2.4.0, 2.5.0, 2.6.0 and 2.7.0 and only 2.4.0 seems to produce the correct output.

**Comment by suparngp** [comment]
I investigated a little bit, This check is failing:

https://github.com/serverless/serverless/blob/master/lib/plugins/aws/package/compile/events/stream/index.js#L274

Because in 2.4.0, the value `true` is read as a boolean where in 2.5.0+, it is read as `'true'`.

```
if (event.stream.consumer && streamType === 'kinesis') {
              console.log(event.stream);
              if (event.stream.consumer === true) {
```

prints

```
// 2.4.0
{
  type: 'kinesis',
  arn: { 'Fn::ImportValue': 'MyServiceStack-Kinesis-events-Arn' },
  consumer: true
}
```

```
// 2.5.0+
{
  type: 'kinesis',
  arn: { 'Fn::ImportValue': 'MyServiceStack-Kinesis-events-Arn' },
  consumer: 'true'
}
```

**Comment by suparngp** [comment]
I found a workaround. If I use `consumer: !!bool true`, it is parsed properly. I am leaving the bug open for someone to review.

**Comment by suparngp** [comment]
I narrowed this bug down to `const ajv = new Ajv({ allErrors: true, coerceTypes: 'array', verbose: true });` in `ConfigSchemaHandler.js`. The `coerceTypes` change was indeed [introduced in v2.5.0](https://github.com/serverless/serverless/compare/v2.4.0...v2.5.0#diff-c00a908c9b817afc737bcf1ea69b5a2f29620c9919030c7348471a8dd695f86dR104) [in this commit](https://github.com/serverless/serverless/pull/8319/files#diff-c00a908c9b817afc737bcf1ea69b5a2f29620c9919030c7348471a8dd695f86dR94).

@medikoo I also found a bug https://github.com/ajv-validator/ajv/issues/1294 that you opened with `ajv` but I did not understand if there is a plan to fix it. Any comment?

**Comment by medikoo** [comment]
@suparngp great thanks for reporting this. Indeed there's a bug in schema.

There's technically no real bug in AJV (rather design issue). Having `coerceTypes: 'array'` is convenient, but it requires double care in defining schema, e.g. when relying on `oneOf` AJV will inspect all possible options, and imply corecion for applicable cases (but without reverting it on mismatch). This was the case here. Where we allowed `true` and _string_ form. When inspecting the _string_ form AJV simply coerced `true` to `'true'`, and left it that way.

Solution is not use `oneOf` but `anyOf`, as that makes AJV stop as soon as first form is matched. 

Fixed in #8405
